src/network/networkd-network.c

```diff
@@ -150,10 +169,21 @@ static int network_verify(Network *network) {
         if (network->ip_masquerade)
                 network->ip_forward |= ADDRESS_FAMILY_IPV4;
 
-        if (network->mtu > 0 && network->dhcp_use_mtu) {
-                log_warning("%s: MTUBytes= in [Link] section and UseMTU= in [DHCP] section are set. "
-                            "Disabling UseMTU=.", network->filename);
-                network->dhcp_use_mtu = false;
+        network_set_stacked_netdevs_mtu(network);
+
+        if (network->mtu > 0) {
+                if (network->dhcp_use_mtu) {
+                        log_warning("%s: MTUBytes= in [Link] section and UseMTU= in [DHCP] section are set. "
+                                    "Disabling UseMTU=.", network->filename);
+                        network->dhcp_use_mtu = false;
+                }
+
+                if (network->mtu < network->stacked_netdevs_mtu) {
```

Comment:
If we know that the stacked devices require certain MTU, shouldn't we set it, even if this Network has mtu==0 ? I.e. I think this `if` should not be moved one level up, and not be under `if (network->mtu > 0)`.

Reply:
Oh, I see this is done in a way in patch 2/3. But then this means that network->mtu is bumped in some cases, but not in others. It seems nicer to simply bump it always, so that we don't have to remember to do MAX(network->mtu, network->stacked_mtu)` everywhere.

Reply:
If `MTUBytes=` is not specified in .network file, then `network->mtu` is `0`, and in that case, `networkd` should not change mtu if device's mtu is already large enough.

Note that some drivers restart interfaces if mtu is changed. I hope `networkd` can handle that, but I think we should not change mtu if not necessary.

Without changing the current logic, how about introducing a boolean flag e.g. `network->force_change_mtu` which indicates `MTUBytes=` is specified or not? Then, we can drop `stacked_netdevs_mtu` from `Network`.